### PR TITLE
Auto remove deployment

### PR DESCRIPTION
**Pull request by Siegrift**
Closes https://github.com/api3dao/airnode/issues/1188

image

image
image
image
image

### CODE REVIEW

**Review by amarthadan — APPROVED** [review]
Other than the flag name LGTM :+1:

File: packages/airnode-deployer/src/cli/commands.ts [hunk 1]
```diff
@@ -96,14 +100,39 @@ export async function deploy(configPath: string, secretsPath: string, receiptFil
...
   }
 
-  const output = goDeployAirnode.data;
-  const deploymentTimestamp = new Date().toISOString();
+  if (!goDeployAirnode.success) {
+    logger.fail(
+      bold(
+        `Airnode deployment failed due to unexpected errors.\n` +
+          `  It is possible that some resources have been deployed on cloud provider.\n` +
+          `  We are attempting to remove them...\n`
```
Comment (andreogle) [inline packages/airnode-deployer/src/cli/commands.ts:108]: I think we should strongly avoid "we" terminology in messages, or more broadly anything that implies the core team has some kind of admin ability to interfere with deployments. It's misleading and is likely going to cause confusion for users. 

I was looking at the screenshots and noticed this

Comment (Siegrift) [reply packages/airnode-deployer/src/cli/commands.ts:108]: Would:

```
It is possible that some resources have been deployed on cloud provider.
Attempting to remove them...
```

be OK?

Comment (andreogle) [reply packages/airnode-deployer/src/cli/commands.ts:108]: Yeah that's better 👍🏻

**Pull request merged**